src/components/application_manager/src/commands/mobile/system_request.cc

```diff
@@ -103,25 +103,38 @@ class QueryAppsDataValidator {
   }
 
   bool ValidateAppDataAndOsAndLanguagesData() const {
-    const smart_objects::SmartArray* objects_array =
-        data_[json::response].asArray();
+    smart_objects::SmartArray* objects_array = data_[json::response].asArray();
     if (!objects_array) {
       LOG4CXX_WARN(logger_,
                    kQueryAppsValidationFailedPrefix
                        << "QueryApps response is not array.");
       return false;
     }
-    const std::size_t arr_size(objects_array->size());
     SynonymsMap synonyms_map;
-    for (std::size_t idx = 0; idx < arr_size; ++idx) {
-      const smart_objects::SmartObject& app_data = (*objects_array)[idx];
+    bool is_response_have_valid_application = false;
+
+    smart_objects::SmartArray::iterator applications_itt =
+        objects_array->begin();
+
+    for (; applications_itt < objects_array->end(); ++applications_itt) {
+      const smart_objects::SmartObject& app_data = *applications_itt;
+      LOG4CXX_FATAL(logger_, &app_data);
 
       if (!app_data.isValid()) {
         LOG4CXX_WARN(logger_,
                      kQueryAppsValidationFailedPrefix
                          << "Wrong application data in json file.");
         return false;
       }
+
+      if (!FindMandatoryParameters(app_data)) {
+        LOG4CXX_WARN(logger_,
+                     "Application havent some of mandatory parameters. "
+                     "Application will be skipped.");
+        objects_array->erase(applications_itt);
+        continue;
+      }
+
       std::set<std::string> app_ids_set;
       if (!ValidateAppIdAndAppName(app_data, app_ids_set)) {
         return false;
```

Comment:
@Kozoriz  The code in `if (os_type.empty()) ` can be refactored it looks like the second check is redundant.. 
And here ` if (app_data[json::android][json::packageName].asString().length()` json::android can be changed to os_type. This part is not in the diff, but could check please.